server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -2177,6 +2173,7 @@ private UserAccount getUserAccount(String username, String password, Long domain
             if (domain != null) {
                 domainName = domain.getName();
             }
+            userAccount = _userAccountDao.getUserAccount(username, domainId);
```

Comment:
If the authenticator returns true without a user in cloudstack in DB, it will still fail. It only happens at a later stage. There are no new assumptions here. It will continue to work as earlier. If its not that would be a bug. I do not see any issue when running this either from the test cases or manually. if you see any issue report it and it can be addressed.